mmv1/templates/terraform/decoders/firebase_database_instance.go.erb

```diff
@@ -0,0 +1,6 @@
+if v := res["state"]; v == "DELETED" {
+	return nil, nil
+} 
+res["desired_state"] = res["state"]
```

Comment:
I'm curious about this line, do you know if it is needed, perhaps by Terraform to look at diffs? My initial thought was that `desired_state` does not have meaning for the object returned from the API.

Reply:
Ah yep, that makes sense. I also found https://github.com/hashicorp/terraform-provider-google/pull/4797#discussion_r374932691, which mentions imports being another case where you want the state change to happen in a subsequent apply after the import happens. Thanks for clarifying!